src/cls/2pc_queue/cls_2pc_queue_types.h

```diff
@@ -38,13 +41,15 @@ struct cls_2pc_urgent_data
 {
   uint64_t reserved_size{0};   // pending reservations size in bytes
   cls_2pc_reservation::id_t last_id{cls_2pc_reservation::NO_ID}; // last allocated id
+  uint32_t committed_entries{0};
   cls_2pc_reservations reservations; // reservation list (keyed by id)
   bool has_xattrs{false};
 
   void encode(ceph::buffer::list& bl) const {
     ENCODE_START(1, 1, bl);
     encode(reserved_size, bl);
     encode(last_id, bl);
+    encode(committed_entries, bl);
```

Comment:
since this is serialized as wire protocol and into disk, we have to preserve backward compatibility.
for that we need to do couple of things:
* `ENCODE_START(2, 1, bl);` - meaning that we are now at version 2, but we support down to version 1
* move the new entries to the end of the structure, so that if ver1 code is going to read them, they would just ignore them
* in the "decode()" indicate that we are at ver2: `DECODE_START(2, bl);`
* in the "decode()" function you have to decode based on the versions:
```cpp
if (struct_v >= 2) {
  decode(committed_entries, bl);
}
```

Reply:
similar commenst on: `cls_2pc_reservation`

Reply:
Ack!